metadata/pkg/drivers/aws/aws.go

```diff
@@ -43,10 +50,122 @@ type AwsAdapter struct {
 	Session *session.Session
 }
 
+func iterateObjects(i int, object *s3.Object, svc *s3.S3, bucketName string, objectArray []model.MetaObject, wg *sync.WaitGroup) {
```

Comment:
IMO, GetObjectsParams or similar would be suitable name. You may consider changing the name

Reply:
removed go routine at object level. so this function is removed.